src/main/java/com/microsoft/sqlserver/jdbc/SQLServerFMTQuery.java

```diff
@@ -111,3 +115,19 @@ String getFMTQuery() {
         SQLServerParser.parseQuery(iter, this);
     }
 }
+
+
+class SQLServerErrorListener extends BaseErrorListener {
+    static final private java.util.logging.Logger logger = java.util.logging.Logger
+            .getLogger("com.microsoft.sqlserver.jdbc.internals.SQLServerFMTQuery");
+
+    @Override
+    public void syntaxError(Recognizer<?, ?> recognizer, Object offendingSymbol, int line, int charPositionInLine,
+            String msg, RecognitionException e) {
+        if (logger.isLoggable(java.util.logging.Level.FINE)) {
+            logger.fine("Error occured during token parsing: " + msg);

```

Comment:
why not just 1 call?

Reply:
Wanted 2 lines. 1 line for the error message and the 2nd line for what token couldn't be read and where it is.

Reply:
"\n"?

Reply:
Think it shows up a little weird in logs, every line generally has some prefix with timestamp info, etc. There are other places in the driver that double logs like this.

Reply:
okey, no big deal I guess...